lib/xcore.js

```diff
@@ -31,14 +31,20 @@ function XClient(displayNum, screenNum, options)
 
     this.displayNum = displayNum;
     this.screenNum = screenNum;
-    this.authHost = os.hostname();
 }
 util.inherits(XClient, EventEmitter);
 
 XClient.prototype.init = function(stream)
 {
     this.stream = stream;
 
+    this.authHost = stream.remoteAddress;
+    this.authFamily = stream.remoteFamily;
+    if (!this.authHost || this.authHost === '127.0.0.1' || this.authHost === '::1') {
+      this.authHost = os.hostname();
+      this.authFamily = null;
+    }
```

Comment:
I mean the xauthority file format? Or just looking to xauth implementation should do?